txrelayer/txrelayer.go

```diff
@@ -61,14 +64,25 @@ type TxRelayerImpl struct {
 
 func NewTxRelayer(opts ...TxRelayerOption) (TxRelayer, error) {
 	t := &TxRelayerImpl{
-		ipAddress:      DefaultRPCAddress,
-		receiptTimeout: 50 * time.Millisecond,
-		numRetries:     defaultNumRetries,
+		ipAddress:        DefaultRPCAddress,
+		receiptsPollFreq: DefaultPollFreq,
+		receiptsTimeout:  DefaultTimeoutTransactions,
 	}
 	for _, opt := range opts {
 		opt(t)
 	}
 
+	// Calculate receiptsPollFreq based on receiptsTimeout
+	if t.receiptsTimeout >= time.Minute {
+		t.receiptsPollFreq = 2 * time.Second
+	}
```

Comment:
Why do we (over-)complicate things with such checks? If there aren't any particular reason for it to be here, I'd remove it.


```suggestion
```